### PR TITLE
CES-60 mark recurring meeting as completed

### PR DESCRIPTION
**Pull request by VuHoDac**
https://issues.liferay.com/browse/CES-60

### CODE REVIEW

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java [hunk 1]
```diff
@@ -222,7 +222,18 @@ public void endPowwowMeeting(
 		long powwowMeetingId = ParamUtil.getLong(
 			actionRequest, "powwowMeetingId");
 
+		PowwowMeeting powwowMeeting =
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java:225]: we can get the meeting instance from `PowwowServiceProviderUtil.endPowwowMeeting`

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java [hunk 2]
```diff
@@ -222,7 +222,18 @@ public void endPowwowMeeting(
...
+			PowwowMeetingServiceUtil.getPowwowMeeting(powwowMeetingId);
+
 		PowwowServiceProviderUtil.endPowwowMeeting(powwowMeetingId);
+
+		boolean isUpdatedCompleted = Validator.isNull(powwowMeeting.findNextOccurrence());
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java]: should rename variable `isUpdatedCompleted` to `hasNextOccurrence` with:
`boolean hasNextOccurrence = Validator.isNotNull(powwowMeeting.findNextOccurrence());`

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java [hunk 3]
```diff
@@ -222,7 +222,18 @@ public void endPowwowMeeting(
...
+
+		PowwowMeetingLocalServiceUtil
+			.updateStatus(
+				powwowMeetingId,
+				isUpdatedCompleted ? PowwowMeetingConstants.STATUS_COMPLETED
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java]: extract the updating status to a varibale

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/service/impl/PowwowMeetingLocalServiceImpl.java [hunk 4]
```diff
@@ -123,13 +123,15 @@ public void checkPowwowMeetings() throws PortalException {
...
 			}
 
-			if (!PowwowServiceProviderUtil.isPowwowMeetingRunning(
-					powwowMeeting.getPowwowMeetingId())) {
-
-				updateStatus(
-					powwowMeeting.getPowwowMeetingId(),
-					PowwowMeetingConstants.STATUS_COMPLETED);
-			}
+			boolean isUpdatedCompleted =
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/service/impl/PowwowMeetingLocalServiceImpl.java]: rename this variable, and make the logic easier to understand

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/service/impl/PowwowMeetingLocalServiceImpl.java [hunk 5]
```diff
@@ -123,13 +123,15 @@ public void checkPowwowMeetings() throws PortalException {
...
+				!PowwowServiceProviderUtil.isPowwowMeetingRunning(
+					powwowMeeting.getPowwowMeetingId())
+						&& Validator.isNull(powwowMeeting.findNextOccurrence());
+
+			updateStatus(
+				powwowMeeting.getPowwowMeetingId(),
+				isUpdatedCompleted ? PowwowMeetingConstants.STATUS_COMPLETED
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/service/impl/PowwowMeetingLocalServiceImpl.java]: extract the updating status to a variable

**Review by hhtank — CHANGES_REQUESTED** [review]

**Comment by VuHoDac** [comment]
updated

File: modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/service/impl/PowwowMeetingLocalServiceImpl.java [hunk 6]
```diff
@@ -123,13 +123,16 @@ public void checkPowwowMeetings() throws PortalException {
 				continue;
 			}
 
-			if (!PowwowServiceProviderUtil.isPowwowMeetingRunning(
-					powwowMeeting.getPowwowMeetingId())) {
+			boolean isPowwowMeetingAndNextOccurrence=
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/service/impl/PowwowMeetingLocalServiceImpl.java:126]: this still should be more clear. the original logic was: if meeting is running, then keep it in progress status.

**Review by hhtank — COMMENTED** [review]
merged with a bit correction. thanks!

**Pull request closed without merging**